debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/sink/SinkConnection.java

```diff
@@ -0,0 +1,228 @@
+import com.mongodb.client.MongoClients;
+import com.mongodb.event.ClusterClosedEvent;
+import com.mongodb.event.ClusterDescriptionChangedEvent;
+import com.mongodb.event.ClusterListener;
+import com.mongodb.event.ClusterOpeningEvent;
+
+import io.debezium.config.Field;
+
+public final class SinkConnection {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SinkConnection.class);
+    private static final String USERS_INFO = "{usersInfo: '%s', showPrivileges: true}";
+    private static final String ROLES_INFO = "{rolesInfo: '%s', showPrivileges: true, showBuiltinRoles: true}";
+
+    public static Optional<MongoClient> canConnect(final Config config, final Field connectionStringConfigName) {
+        Optional<ConfigValue> optionalConnectionString = getConfigByName(config, connectionStringConfigName.name());
+        if (optionalConnectionString.isPresent() && optionalConnectionString.get().errorMessages().isEmpty()) {
+            ConfigValue configValue = optionalConnectionString.get();
+
+            AtomicBoolean connected = new AtomicBoolean();
+            CountDownLatch latch = new CountDownLatch(1);
+            ConnectionString connectionString = new ConnectionString((String) configValue.value());
+            MongoClientSettings mongoClientSettings = MongoClientSettings.builder()
+                    .applyConnectionString(connectionString)
+                    .applyToClusterSettings(
+                            b -> b.addClusterListener(
+                                    new ClusterListener() {
+                                        @Override
+                                        public void clusterOpening(final ClusterOpeningEvent event) {
+                                        }
+
+                                        @Override
+                                        public void clusterClosed(final ClusterClosedEvent event) {
+                                        }
+
+                                        @Override
+                                        public void clusterDescriptionChanged(
+                                                                              final ClusterDescriptionChangedEvent event) {
+                                            ReadPreference readPreference = connectionString.getReadPreference() != null
+                                                    ? connectionString.getReadPreference()
+                                                    : ReadPreference.primaryPreferred();
+                                            if (!connected.get()
+                                                    && event.getNewDescription().hasReadableServer(readPreference)) {
+                                                connected.set(true);
+                                                latch.countDown();
+                                            }
+                                        }
+                                    }))
+                    .build();
```

Comment:
I saw this when I studied other Mongo DB connectors. I think this is meant for MongoDB Atlas clusters and handling that nodes get removed or added? I don't know and not like M much, so I am not 100% sure.

Reply:
I can imagine the use in cases, where you need to e.g. restart connection when cluster topology would change. I intended to use something very similar for killing change stream in case of a promotion event.  

However here it seems to be a one time use -- meaning any attempt to communicate with the database would do the same.

Regardless, the information that other use it is valuable and  I will try to have a look at what e.g. the official Mongodb connector does and why they would use that in case such as this one. 

BTW: I haven't personally verified (it is on my toto list), however based on what Bobby told me, the event is not fired in case of added replica to an existing shard.

Reply:
Ok, so this seems to be done this way primarily for two reason 

1) At this point we don't know what permissions we are missing (so forcing the connection with e.g. listing databases might not work)
2) It somewhat has the benefit of not really interacting with the database any more than we need to. 

We should probably utilise similar approach even in source.

Reply:
Agreed. I will file a Jira for it!

Reply:
added to https://issues.redhat.com/browse/DBZ-7995